src/shogun/classifier/svm/LibLinear.cpp

```diff
@@ -117,7 +117,7 @@ bool LibLinear::train_machine(std::shared_ptr<Features> data)
 	}
 	SGVector<float64_t> w;
 	if (get_bias_enabled())
-		w = SGVector<float64_t>(SG_MALLOC(float64_t, num_feat + 1), num_feat);
+		w = SGVector<float64_t>(num_feat + 1, num_feat);
```

Comment:
mmm this one is actually a bit tricky.... since the idea is that the last element of W is the bias... but when u actually do most of the operations with W you actually want to have it's length to be the feat space dim... i.e. w.vlen should be num_feat but the size of w should be num_feat + 1 and the last element of w is used as the bias... from mem management perspective, apart from what's the size of the chunk u allocate the vlen doesn't matter... in other words:
```
w = SGVector<float64_t>(num_feat + 1);
w.vlen = num_feat;
```
could work... or
```
w = SGVector<float64_t>(SG_ALIGNED_ALLOC(num_feat + 1), num_feat);
```

the former is maybe easier to understand (although imo that vlen should be a const vlen and then this wouldn't be possible) and really follows the encapsulation concept as you dont expose the internal workings of the SGVector container. the latter is just a simple patch, but of course this breaks encapsulation, hence if tomorrow somebody refactors the whole SGVector internals this definitely need to be handled as a corner case (i.e. directly calling SG_ALIGNED_MALLOC).

Reply:
Okay so which one should I go for in this case

Reply:
@gf712 any opinion about this?

Reply:
I think you should either use second one, or rewrite the algo so that you don’t have to use this hack... I agree that you shouldn’t change vlen, at least via the public interface

Reply:
actually `w = SGVector<float64_t>(num_feat + 1);` should work fine, you just need to change `set_w` to use a slice of the vector, i.e. `m_w=w[0:num_feat+1]` and `b = [num_feat+1]`. @vigsterkr from what i can see the number of features in the algo comes from the `prob` object and not SGVector.